monai/data/dataset.py

```diff
@@ -807,6 +821,20 @@ def set_data(self, data: Sequence):
         def _compute_cache_num(data_len: int):
             self.cache_num = min(int(self.set_num), int(data_len * self.set_rate), data_len)
 
+        def _compute_cache(indices=None):
+            if self.runtime_cache or (self._is_dist and dist.get_rank() != self.cache_rank):
+                cache = [None for i in range(self.cache_num)]
+            else:
+                cache = self._fill_cache(indices)
+            if self.runtime_cache:
+                cache = self._set_multiprocessing_cache(cache, mp_cache=True)
+
+            if self._is_dist:
+                obj_list = [cache]
+                dist.broadcast_object_list(obj_list, src=self.cache_rank)
+                cache = obj_list[0]
+            return cache
+
```

Comment:
- so here, if we don't do runtime_cache , but is_dist==True and rank!=cache_rank,  the cache list will be all Nones (without precomputing). I don't think it's correct. 
When runtime_cache==False, we should always precompute on all ranks.  

- we can remove cache_rank user option.
- check for broadcast should be only under runtime_cache, since we use shared list only in that case

I'd change to:
 ```
  def _compute_cache(indices=None):

    if self.runtime_cache:
        cache = self._set_multiprocessing_cache([None,]*self.cache_num, mp_cache=True)
        if self._is_dist:
            obj_list = [cache]
            dist.broadcast_object_list(obj_list)
            cache = obj_list[0]

    else:
        cache = self._fill_cache(indices)
   
    return cache
```

Reply:
Hi @myron ,

For the "not runtime cache" case, I think you want to broadcast the precomputed cache content to every rank to avoid duplicated computation and cache, is it right?

Thanks.

Reply:
@Nic-Ma no, here we're only broadcasting the **Empty** ProxyList (pointer), so that all processes know where to read/write to shared cache.  This broadcasting should be done before any data caching begins

Reply:
Updated to only broadcast for the `runtime cache` case.

Thanks.